collector/rpc_metrics.go

```diff
@@ -142,25 +181,36 @@ func (collector *NodeRpcMetrics) Describe(ch chan<- *prometheus.Desc) {
 	ch <- collector.blockLagDesc
 	ch <- collector.blocksMissedDesc
 	ch <- collector.syncingDesc
+	ch <- collector.internalStatusRPCFail
+	ch <- collector.internalStatusRPCSuccess
+	ch <- collector.externalStatusRPCFail
+	ch <- collector.externalStatusRPCSuccess
 	ch <- collector.versionBuildDesc
 	ch <- collector.validatorRPCFail
 	ch <- collector.validatorRPCSuccess
 	ch <- collector.prevEpochKickoutDesc
 }
 
 func (collector *NodeRpcMetrics) Collect(ch chan<- prometheus.Metric) {
-	sr, err := collector.internalClient.Get("status", nil)
-	if err != nil {
-		ch <- prometheus.NewInvalidMetric(collector.versionBuildDesc, err)
-		return
-	}
+	fmt.Println("metrics collect")
 
-	srExt, err := collector.externalClient.Get("status", nil)
-	if err != nil {
-		ch <- prometheus.NewInvalidMetric(collector.versionBuildDesc, err)
+	ch <- prometheus.MustNewConstMetric(collector.validatorRPCFail, prometheus.GaugeValue, float64(collector.validatorRPCFailCount))
+	ch <- prometheus.MustNewConstMetric(collector.validatorRPCSuccess, prometheus.GaugeValue, float64(collector.validatorRPCSuccessCount))
+	ch <- prometheus.MustNewConstMetric(collector.internalStatusRPCFail, prometheus.GaugeValue, float64(collector.internalStatusRPCFailCount))
+	ch <- prometheus.MustNewConstMetric(collector.internalStatusRPCSuccess, prometheus.GaugeValue, float64(collector.internalStatusRPCSuccessCount))
+	ch <- prometheus.MustNewConstMetric(collector.externalStatusRPCFail, prometheus.GaugeValue, float64(collector.externalStatusRPCFailCount))
+	ch <- prometheus.MustNewConstMetric(collector.externalStatusRPCSuccess, prometheus.GaugeValue, float64(collector.externalStatusRPCSuccessCount))
+
+	if collector.externalStatusRPCSuccessCount == 0 ||
+		collector.internalStatusRPCSuccessCount == 0 ||
+		collector.validatorRPCSuccessCount == 0 {
+		fmt.Println("Not enough data to start reporting metrics")
 		return
 	}
 
+	sr := collector.currentInternalStatusData
+	srExt := collector.currentExternalStatusData
```

Comment:
what's `sr` stand for in relation to `collector.currentInternalStatusData`?

Would prefer if we could use a more verbose name

Reply:
agree